File: root/common/src/main/java/org/sc/common/rest/PoiResponse.java
package org.sc.common.rest;

import java.util.Collections;
import java.util.List;
import java.util.Set;

public class PoiResponse extends RESTResponse {
    final List<PoiDto> pois;

    public PoiResponse(List<PoiDto> pois, Status status, Set<String> messages) {
        super(status, messages);
        this.pois = pois;
    }

    public PoiResponse(List<PoiDto> pois) {
        super(Status.OK, Collections.emptySet());
        this.pois = pois;
    }

    public List<PoiDto> getPois() {
        return pois;
    }
}

File: root/rest/src/main/java/org/sc/common/rest/response/PoiResponse.kt
package org.sc.common.rest.response

import org.sc.common.rest.PoiDto
import org.sc.common.rest.Status

data class PoiResponse(
    val status: Status,
    val messages: Set<String>,
    val content: List<PoiDto>,
    override val currentPage: Long,
    override val totalPages: Long,
    override val size: Long,
    override val totalCount: Long
) :
    RESTResponse(currentPage, totalPages, size, totalCount)
